api/src/lib.rs

```diff
@@ -571,11 +572,14 @@ impl Client {
         debug!("Attempting GET `{}`", url);
         let http_response = self
             .with_retries(|| {
-                self.http_client
+                let mut request = self
+                    .http_client
                     .get(url.clone())
-                    .headers(self.headers.clone())
-                    .query(query)
-                    .send()
+                    .headers(self.headers.clone());
+                if let Some(query) = query {
+                    request = request.query(query);
+                }
+                request.send()
             })
             .chain_err(|| ErrorKind::Unknown {
                 message: "GET operation failed.".to_owned(),
```

Comment:
If you're wanting to shoot for maximum code reuse, this same structure is basically mirrored in the first part of `delete_query` barring `http_client.get` vs `http_client.delete` - could potentially use `http_client.request` to parametrise this bit over the method.

But I'm also fine with leaving this as-is :smile:

Reply:
I thought about that, but decided to leave as is :sweat_smile: if we need another `*_query` handler I'll definitely factor it out